src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php

```diff
@@ -205,11 +210,11 @@ public function create_item_permissions_check( $request ) {
 	 */
 	public function create_item( $request ) {
 
-		if ( ! defined( 'DOING_AUTOSAVE' ) ) {
+		if ( ! defined( 'WP_RUN_CORE_TESTS' ) && ! defined( 'DOING_AUTOSAVE' ) ) {
```

Comment:
Which specifically? normally we would want DOING_AUTOSAVE false here

Reply:
@adamsilverstein The XMLRPC tests that created revisions were failing. See https://github.com/WordPress/wordpress-develop/actions/runs/6399705846/job/17372187021#step:14:344

This is because if `DOING_AUTOSAVE` is defined that revisions are not created.

Reply:
If this can't be resolved in source, it may be worth looking at these tests and the XML-RPC tests, and deciding whether one of these need to run in a separate process. Note that this will slow down those specific tests, but that's par for the course if there's no reasonable alternative.

If all the tests in a class need to define `DOING_AUTOSAVE`, then the `@runTestsInSeparateProcess` annotation can be added to the test class docblock [Reference](https://docs.phpunit.de/en/9.6/annotations.html#runtestsinseparateprocesses). If it's only needed for individual test methods, `@runInSeparateProcess` in the test methods' docblocks is the appropriate annotation [Reference](https://docs.phpunit.de/en/9.6/annotations.html#runinseparateprocess).

In either case, the `@preserveGlobalState disabled` annotation will also be required to avoid an Exception relating to the serialization of `'Closure'` [Reference](https://docs.phpunit.de/en/9.6/annotations.html#preserveglobalstate).

These annotations are used elsewhere in the test suite when constants and such need to be defined. I believe there may even be some REST API tests that do this, so it may be worth a search for these annotations in the REST API and XML-RPC tests if you want to make sure there's precedent for similar scenarios.

Whichever of the tests that need to run in a separate process can then do `define( 'DOING_AUTOSAVE', true/false );` as needed. If it's needed for all tests in a test class, this can probably just go in `set_up_before_class()/wpSetUpBeforeClass()`, whichever of these methods already exists in the test class.